scripts/core/sampler.yml

```diff
@@ -155,10 +155,10 @@ params:
       desc: "[in] size in bytes of the sampler property value provided"
     - type: "void*"
       name: pPropValue
-      desc: "[out][typename(propName, propSize)] value of the sampler property"
+      desc: "[out][typename(propName, propSize)][optional] value of the sampler property"
     - type: "size_t*"
       name: pPropSizeRet
-      desc: "[out] size in bytes returned in sampler property value"
+      desc: "[out][optional] size in bytes returned in sampler property value"
```

Comment:
While these are optional, they still have valid usage requirements that we should define:

```cpp
if (propSize == 0 && pPropValue != nullptr) {
  return UR_RESULT_ERROR_INVALID_SIZE;
}
if (propSize != 0 && pPropValue == nullptr) {
  return UR_RESULT_ERROR_INVALID_NULl_POINTER;
}
if (pPropValue == nullptr && pPropSizeRet == nullptr) {
  return UR_RESULT_ERROR_INVALID_NULl_POINTER;
}
```

I wonder if this also needs applied to all the other `*GetInfo` entry points...

Reply:
I have fixed this for sampler, but I've created a separate issue to update all remaining `GetInfo` style entry-points with these checks - #549